[#153091168] Fix Dockerfile now bower is removed

46bit opened the pull request:
## What

Until this PR, the `Dockerfile` has expected a `bower.json` file to exist. We deleted it two months ago because the product page app no longer needs it. However, this meant that the DockerHub docker image builds started to fail--and have been doing so ever since.

Additionally, this PR makes the testsuite [be run before deploying from the release CI](https://concourse.build.ci.cloudpipeline.digital/teams/main/pipelines/paas-product-page/jobs/test/builds/32).

## How to review

Code review should suffice. We've run the tests in the new `Dockerfile` successfully with something like `docker run -v /Users/michaelmokrysz/gds/paas-product-page:/paas-product-page fbc6fb9c22ee /bin/sh -c "cd paas-product-page; make test"`.

## Who can review

Not @46bit @camelpunch

alext commented:
Given we already run the tests in Travis, I don't think there's a reason to also run them on the build concourse.

Can you also describe what was broken about the Dockerfile and why it needed fixing?

Commit:
Removed bower dependency from the Dockerfile

Docker builds have been broken since we removed the `bower.json` file.
This commit also compiles `make` into the Docker image so that we can
run the test suite from the Docker image.

Commit:
Run tests when deploying from Concourse

Our build Concourse has app deploy pipelines, including one for the
product page [1]. We have the option of running the tests before we
deploy, to prevent deployment of a broken build. This commit uses this
option.

[1] https://concourse.build.ci.cloudpipeline.digital/teams/main/pipelines/paas-product-page
